src/encoding.rs

```diff
@@ -55,25 +52,17 @@ impl Decoder {
 
     fn store_and_decoded_data<'a>(&mut self, data: &'a [u8]) -> Option<(&[u8], &'a [u8])> {
         // Process frame header
-        let (expected_size, data) = match decode_size(&self.stored) {
+        let ((expected_size, used_bytes), data) = match decode_size(&self.stored) {
             Some(size) => (size, data),
             None => {
-                let remaining = PADDING - self.stored.len();
-                if data.len() >= remaining {
```

Comment:
What would happen in the case where the currently stored data + the new input data are enough to know the size? The Decoder will skip that case.
Example: 
Message of 500 bytes: (2 size + 498 payload)
1. **Read action**: Decoder read 1 byte. (nothing to decode)
1. **Read action**. Decoder reads 499 bytes. (nothing to decode when there is!)

The test `decode_message_after_non_enough_padding` check that behavior but is broken know because the message is really small and its size fit in one byte. I think that increasing the message size in that tests makes it works again.

The same applies to the `decode_byte_per_byte` test, the message should be increased to test correctly

Reply:
Will fix!